### ISSUE TITLE
latest source has compile issues on freebsd

### ISSUE THREAD

**Issue by danielb2** [fish-shell/fish-shell#2615]
Compiling commit aefe08412454074ddc3a85e98e9449c2dfc8830f

Dunno if this is different but here it is.

```bash
[~/t/fish-shell]$ cat FISH-BUILD-VERSION-FILE
FISH_BUILD_VERSION = 2.2.0-465-gaefe084
```

Any thoughts of what the issue is here? I did a `gmake clean` and started from scratch but still has issues.

``` bash
g++ -g -O2 -D_LARGEFILE_SOURCE=1 -D_FILE_OFFSET_BITS=64 -fno-exceptions -Wall -Wno-sign-compare -I/usr/local/include -iquote. -iquote./src/ -DLOCALEDIR=\"/usr/local/share/locale\" -DPREFIX=L\"/usr/local\" -DDATADIR=L\"/usr/local/share\" -DSYSCONFDIR=L\"/usr/local/etc\" -DBINDIR=L\"/usr/local/bin\" -DDOCDIR=L\"/usr/local/share/doc/fish\"  -DFISH_BUILD_VERSION=\"2.2.0-465-gaefe084\"   -rdynamic obj/function.o obj/builtin.o obj/complete.o obj/env.o obj/exec.o obj/expand.o obj/highlight.o obj/history.o obj/kill.o obj/parser.o obj/proc.o obj/reader.o obj/sanity.o obj/tokenizer.o obj/wildcard.o obj/wgetopt.o obj/wutil.o obj/input.o obj/output.o obj/intern.o obj/env_universal_common.o obj/input_common.o obj/event.o obj/signal.o obj/io.o obj/parse_util.o obj/common.o obj/screen.o obj/path.o obj/autoload.o obj/parser_keywords.o obj/iothread.o obj/color.o obj/postfork.o obj/builtin_string.o obj/builtin_test.o obj/parse_tree.o obj/parse_productions.o obj/parse_execution.o obj/pager.o obj/utf8.o obj/fish_version.o obj/wcstringutil.o obj/fish.o -lncurses -lpthread  -L/usr/local/lib -Wl,-R/usr/local/lib -lpcre2-32 -o fish
obj/common.o: In function `show_stackframe()':
/usr/home/daniel/tmp/fish-shell/src/common.cpp:94: undefined reference to `backtrace_symbols_fd'
c++: error: linker command failed with exit code 1 (use -v to see invocation)
Makefile:796: recipe for target 'fish' failed
gmake: *** [fish] Error 1
```

**Comment by faho** [comment]
You might need [libexecinfo](http://www.freebsdsoftware.org/devel/libexecinfo.html). Can you try that?

**Comment by zanchey** [comment]
I can reproduce the problem, and it's fixed with `./configure LDFLAGS='-lexecinfo'`.

ea798011b25 switched from `backtrace_symbols` to `backtrace_symbols_fd`, but the configure script and fallback implementations didn't get updated.

**Comment by danielb2** [comment]
I've confirmed @zanchey's statement. This does make it work. Thanks @zanchey

**Comment by danielb2** [comment]
the fish executable doesn't seem to accept CR, but I guess that's a different issue ...

**Comment by zanchey** [comment]
I think that's due to the change #2578; the default bindings should continue to work but custom bindings may need updating. If you have any concerns that PR is the place to start.

**Comment by krader1961** [comment]
> the fish with this version doesn't seem to accept CR, but I guess that's a different issue ...
 
As @zanchey noted that's almost certainly due to my PR #2578. If you're running a fish binary freshly built from source but did not install the accompanying scripts then CR won't work. You can press [ctrl-J] (aka LF) to enter commands. It's also possible, but less likely, that you have custom key bindings which have bound \cM (aka \r) to something other than `execute`. Try running

```
bind | grep execute[ctrl-J]
```

If you don't see bindings for both \n and \r then you haven't installed the updated share/functions/fish_default_key_bindings.fish file.

**Comment by ridiculousfish** [comment]
I guess we need to update the configure script. Are there platforms without `backtrace_symbols_fd`? Is libexecinfo standard issue?

**Comment by zanchey** [comment]
I'll push a patch shortly.

**Comment by jfrid** [comment]
I just tried to compile the latest commit on master in Cygwin but it fails on `backtrace_symbols_fd`. From my googling, it seems that there is no support for `backtrace_symbols_fd` in Cygwin yet. This is the only attempt at implementing it I've found so far: https://cygwin.com/ml/cygwin-developers/2015-03/msg00005.html

Is there a workaround for Cygwin using e.g. libexecinfo? It seems to be missing also.
